### ISSUE TITLE
Adding a method to get node httpServer instance

### ISSUE THREAD

**Issue by OneCalmCloud** [nitrojs/nitro#2040]
### Describe the feature

By referencing Nitro's HTTP server, it can be possible to enable WebSocket and Web Server to share the same port without the need for reverse proxy.

Related: [Nuxt.js websocket-servers](https://github.com/nuxt/nuxt/blob/00d2b90ea074d2cc23393610f5ddcb9943421745/docs/2.guide/3.going-further/12.websocket-servers.md) and [how-to-fix-socket-io-with-nitro-server-in-nuxtjs-3](https://stackoverflow.com/questions/77298957/how-to-fix-socket-io-with-nitro-server-in-nuxtjs-3)

As long as there is a method to obtain Nitro's HTTP server, initializing a Socket.IO instance can be done as follows:

- https://github.com/unjs/nitro/blob/main/src/runtime/entries/nitro-dev.ts#L20
- https://github.com/unjs/nitro/blob/main/src/runtime/entries/node-server.ts#L15-L18

Implementation Methods:
1. Use Nitro Runtime Hooks
   Add a hook similar to the one mentioned in [how-to-fix-socket-io-with-nitro-server-in-nuxtjs-3](https://stackoverflow.com/questions/77298957/how-to-fix-socket-io-with-nitro-server-in-nuxtjs-3). In the callback function, access the `httpServer`.

2. Create a Global Function `useAppServer`
   Return the `httpServer` object by using `useAppServer()`.
  
Just like:
   ```javascript
// plugins/socketIO.ts
import { Server } from "socket.io";

export default defineNitroPlugin(async (nitroApp) => {
  try {
    nitroApp.hooks('server', (server) => {
      const io = new Server(server, {
        cors: {
          origin: "*",
        },
      });
      io.on("connection", (socket) => {
        console.log("A user has connected");
        socket.on("disconnect", function () {
          console.log("A user has disconnected");
        });
      });
    })

    console.log("Socket.IO is running");
  } catch (error) {
    console.log("Socket.IO connection failed.", error.message);
  }
});
```
or
```javascript
// plugins/socketIO.ts
import { Server } from "socket.io";

export default defineNitroPlugin(async (nitroApp) => {
  try {
    const server = useAppServer();
    const io = new Server(server, {
      cors: {
        origin: "*",
      },
    });
    io.on("connection", (socket) => {
      console.log("A user has connected");
      socket.on("disconnect", function () {
        console.log("A user has disconnected");
      });
    });

    console.log("Socket.IO is running");
  } catch (error) {
    console.log("Socket.IO connection failed.", error.message);
  }
});
```

### Additional information

- [x] Would you be willing to help implement this feature?

```[tasklist]
### Tasks
```

**Comment by melasculla** [comment]
@pi0 So you suggest to use built in websocket rather than socket.io?

**Comment by melasculla** [comment]
https://nuxt-chat.pi0.io/ chat demo doesn't work btw

**Comment by pi0** [comment]
@melasculla yes (unless you have reasons to not) ! built-in websocket is not custom protocol and works for all runtimes (not limited to Node.js only)

**Comment by melasculla** [comment]
@pi0 I see, testing it rn, but I'm a bit lost, how can I setup auth?, using `peer.request.headers` and parsing em manually it's only way?

Moved to https://github.com/nitrojs/nitro/discussions/3408
